### ISSUE TITLE
Replace atoi usages with strtol

### ISSUE THREAD

**Issue by Lemongrass3110** [rathena/rathena#1609]
Currently our code uses atoi way to often, which is considered unsafe and might result in unexpected behaviours.

We have a total of 865 usages in 47 files:
- [ ] src\char\char.c (51 hits)
- [ ] src\char\char_clif.c (1 hit)
- [ ] src\char\char_logif.c (3 hits)
- [ ] src\char\char_mapif.c (12 hits)
- [ ] src\char\inter.c (16 hits)
- [ ] src\char\int_auction.c (15 hits)
- [ ] src\char\int_elemental.c (15 hits)
- [ ] src\char\int_guild.c (38 hits)
- [ ] src\char\int_homun.c (22 hits)
- [ ] src\char\int_mail.c (31 hits)
- [ ] src\char\int_mercenary.c (12 hits)
- [ ] src\char\int_party.c (12 hits)
- [ ] src\char\int_pet.c (10 hits)
- [ ] src\char\int_storage.c (25 hits)
- [ ] src\common\core.c (2 hits)
- [ ] src\common\msg_conf.c (1 hit)
- [ ] src\common\socket.c (4 hits)
- [ ] src\common\sql.c (2 hits)
- [ ] src\common\timer.c (1 hit)
- [ ] src\login\account.c (8 hits)
- [ ] src\login\ipban.c (4 hits)
- [ ] src\login\login.c (14 hits)
- [ ] src\map\atcommand.c (67 hits)
- [ ] src\map\battle.c (1 hit)
- [ ] src\map\buyingstore.c (11 hits)
- [ ] src\map\cashshop.c (4 hits)
- [ ] src\map\channel.c (1 hit)
- [ ] src\map\clif.c (6 hits)
- [ ] src\map\elemental.c (29 hits)
- [ ] src\map\guild.c (5 hits)
- [ ] src\map\homunculus.c (58 hits)
- [ ] src\map\instance.c (5 hits)
- [ ] src\map\itemdb.c (66 hits)
- [ ] src\map\log.c (4 hits)
- [ ] src\map\map.c (11 hits)
- [ ] src\map\mapreg.c (1 hit)
- [ ] src\map\mercenary.c (27 hits)
- [ ] src\map\mob.c (68 hits)
- [ ] src\map\npc.c (9 hits)
- [ ] src\map\pc.c (47 hits)
- [ ] src\map\pc.h (1 hit)
- [ ] src\map\pet.c (18 hits)
- [ ] src\map\quest.c (7 hits)
- [ ] src\map\script.c (6 hits)
- [ ] src\map\skill.c (94 hits)
- [ ] src\map\status.c (10 hits)
- [ ] src\map\vending.c (10 hits)

Desired sample usage it should be replace with would look like this:
```
int value;
char* validation_pointer, buffer[X];

value = strtol( buffer, &validation_pointer, 10 ); // or 16 if you want hexadecimal

if( validation_pointer != NULL ){
 // TODO: add error handling here
}
```

**Comment by secretdataz** [comment]
This could be cleaned up in the C++ branch.

**Comment by lighta** [comment]
Good idea, I was afraid about the exception for a moment but it seem it has nothrow garantie so it's all good.
